src/Magnum/EigenIntegration/Integration.h

```diff
@@ -369,6 +370,51 @@ template<class To, std::size_t size, class T> inline To cast(const Math::Vector<
     return Math::Implementation::VectorConverter<size, T, To>::to(from);
 }
 
+
+/**
+@brief Convert Corrades StridedArrayView2D to Eigens Dynamic Matrix Type.
+*/
+template<class T> inline Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>, Eigen::Unaligned, Eigen::Stride<Eigen::Dynamic, Eigen::Dynamic>> arrayCast(const Containers::StridedArrayView2D<T>& from) {
+    using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+    using StrideT = Eigen::Stride<Eigen::Dynamic, Eigen::Dynamic>;
+    auto size = from.size();
+    auto stride = from.stride();
+    return Eigen::Map<MatrixT, Eigen::Unaligned, StrideT>(reinterpret_cast<T*>(from.data()), size[0], size[1], StrideT(stride[0]/sizeof(T), stride[1]/sizeof(T)));
+}
+
+/**
+@brief Convert Corrades StridedArrayView1D to Eigens Dynamic Vector Type.
+Since for a one dimensional StridedArrayView there is no column or row version
+we always return an Eigen Column Vector.
+*/
+template<class T> inline Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, 1>, Eigen::Unaligned, Eigen::InnerStride<>> arrayCast(const Containers::StridedArrayView1D<T>& from) {
+    using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, 1>;
+    return Eigen::Map<MatrixT, Eigen::Unaligned, Eigen::InnerStride<>>(reinterpret_cast<T*>(from.data()), from.size(), Eigen::InnerStride<>(from.stride()/sizeof(T)));
+}
+
+/**
+@brief Convert an Eigen Expression to Corrades StridedArrayView1D
+If it is known at compile time that the Eigen Expression has either
+one Column or one Row, then this function maps the Eigen Expression
+to a one dimensional StridedArrayView.
+*/
+template<class Derived> inline typename std::enable_if<(Eigen::internal::traits<Derived>::ColsAtCompileTime == 1) || (Eigen::internal::traits<Derived>::RowsAtCompileTime == 1), Containers::StridedArrayView1D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
+    using Scalar = typename Derived::Scalar;
+    return Containers::StridedArrayView1D<Scalar>({const_cast<Scalar*>(&from(0,0)), std::size_t(from.size()*from.innerStride())}, std::size_t(from.size()), std::size_t(from.innerStride())*sizeof(Scalar));
+}
+
+/**
+@brief Convert an Eigen Expression to Corrades StridedArrayView2D
+Overload that takes care of any Eigen Expression that was not handled by @ref Magnum::EigenIntegration::arrayCast().
+@snippet EigenIntegration.cpp IntegrationStridedArrayView
+*/
+template<class Derived> inline typename std::enable_if<(Eigen::internal::traits<Derived>::ColsAtCompileTime == Eigen::Dynamic) && (Eigen::internal::traits<Derived>::RowsAtCompileTime == Eigen::Dynamic), Containers::StridedArrayView2D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
```

Comment:
```suggestion
template<class Derived> inline typename std::enable_if<Eigen::internal::traits<Derived>::ColsAtCompileTime == Eigen::Dynamic && Eigen::internal::traits<Derived>::RowsAtCompileTime == Eigen::Dynamic, Containers::StridedArrayView2D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
```

The ()s are not necessary, same above. Btw., both functions have the same argument list, how is the compiler not complaining? In my case I always had to move the SFINAE to the argument list (extra defaulted param, e.g.) to make it work.

Reply:
I think for two function templates to be considered equivalent, expressions involving template parameters in their return types have to be equivalent so the compiler has to evaluate the enable_if struct and one of them is always discarded. But good question why this is allowed since it is forbidden to overload on the return type...